cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -370,9 +368,51 @@ func (p *BicepProvider) Destroy(
 		})
 }
 
-func (p *BicepProvider) getResourceGroupsFromDeployment(ctx context.Context) ([]string, error) {
+// latestDeployment finds the most recent deployment for the given environment in the provided scope.
+func (p *BicepProvider) latestDeployment(ctx context.Context, scope infra.Scope) (*armresources.DeploymentExtended, error) {
+	deployments, err := scope.ListDeployments(ctx)
```

Comment:
This may be an existing issue.. Or not one at all -- but is it possible that the latest deployment is ongoing?

Reply:
If I recall correctly, the deployment `Timestamp` field here can represent different things depending on the time of observation:

1. The start deployment time, when first submitted
1. The ongoing deployment time, after submitting
1. The final deployment time, after completion

I'm not 100% sure if I'm correct on that, but it would be great to confirm the expected behavior here.

Reply:
Confirmed that the value for terminal deployments is suitable for giving us the ordering we want here.